src/Symfony/Component/Console/Command/Command.php

```diff
@@ -250,12 +253,22 @@ public function run(InputInterface $input, OutputInterface $output)
 
         $input->validate();
 
-        if ($this->code) {
-            $statusCode = call_user_func($this->code, $input, $output);
-        } else {
-            $statusCode = $this->execute($input, $output);
+        try {
+            if (!$this->lock($output)) {
+                $statusCode = 0;
```

Comment:
that's wrong. If it cannot run the command, it should return a failure exit code